piptools/scripts/compile.py

```diff
@@ -275,8 +275,11 @@ def cli(
         if src_files == ("-",):
             raise click.BadParameter("--output-file is required if input is from stdin")
         # Use default requirements output file if there is a setup.py the source file
-        elif src_files == ("setup.py",):
-            file_name = DEFAULT_REQUIREMENTS_OUTPUT_FILE
+        elif src_files[0].endswith("setup.py"):
+            file_path = ""
+            if os.path.basename(src_files[0]) != src_files[0]:
+                file_path = os.path.dirname(src_files[0])
+            file_name = os.path.join(file_path, DEFAULT_REQUIREMENTS_OUTPUT_FILE)
```

Comment:
I would simplify this to:

```suggestion
        elif os.path.basename(src_files[0]) == "setup.py":
            file_name = os.path.join(os.path.dirname(src_files[0]), DEFAULT_REQUIREMENTS_OUTPUT_FILE)
```